src/utils/CometAdapter.cpp

```diff
@@ -0,0 +1,589 @@
+
+    //
+    // Optional parameters //
+    //
+
+    //Files
+    registerOutputFile_("pin_out", "<file>", "", "Output file - for Percolator input", false);
+    setValidFormats_("pin_out", ListUtils::create<String>("pin"));
+    registerInputFile_("default_params_file", "<file>", "", "Default Comet params file. All parameters of this take precedence. A template file can be generated using comet.exe -p", false, false, ListUtils::create<String>("skipexists"));
+    setValidFormats_("default_params_file", ListUtils::create<String>("txt"));
+    //registerIntOption_("threads", "<num>", 1, "number of threads", false, true);
+
+    //Masses
+    registerDoubleOption_("precursor_mass_tolerance", "<tolerance>", 10.0, "Precursor mass tolerance, peptide_mass_tolerance(Comet)", false, false);
+    // TO DO: change selection menue from numbers to valid strings
+    //registerStringOption_("precursor_error_units", "<unit>", "ppm", "Parent monoisotopic mass error units", true);
+    //vector<String> valid_strings = ListUtils::create<String>("ppm,Da");
+    registerIntOption_("precursor_error_units", "<choice>", 2, "peptide_mass_units (Comet) 0=amu, 1=mmu, 2=ppm", false, false);
+    //registerIntOption_("mass_type_parent", "<num>", 1, "0=average masses, 1=monoisotopic masses", false, true);
+    //registerIntOption_("mass_type_fragment", "<num>", 1, "0=average masses, 1=monoisotopic masses", false, true);
+    //registerIntOption_("precursor_tolerance_type", "<num>", 0, "0=average masses, 1=monoisotopic masses", false, false);
+    registerIntOption_("isotope_error", "<num>", 0, "0=off, 1=on -1/0/1/2/3 (standard C13 error), 2= -8/-4/0/4/8 (for +4/+8 labeling)", false, false);
+
+    //Search Enzyme
+    vector<String> all_enzymes;
+    EnzymesDB::getInstance()->getAllCometNames(all_enzymes);
+    registerStringOption_("enzyme", "<cleavage site>", "Trypsin", "The enzyme used for peptide digestion.", false, false);
+    setValidStrings_("enzyme", all_enzymes);
+    // TO DO: change selection menue from numbers to valid strings
+    registerIntOption_("num_enzyme_termini", "<num>", 2, "1 (semi-digested), 2 (fully digested, default), 8 C-term unspecific , 9 N-term unspecific", false, false);
+    registerIntOption_("allowed_missed_cleavages", "<num>", 1, "Number of possible cleavage sites missed by the enzyme, maximum value is 5; for enzyme search", false, false);
+
+    //Fragment Ions
+    registerDoubleOption_("fragment_bin_tolerance", "<tolerance>", 1.0005, "fragment_mass_tolerance (MSGF+), fragment_bin_tol (Comet)", false, true);
+    registerDoubleOption_("fragment_bin_offset", "<tolerance>", 0.25, "fragment_bin_offset (Comet)", false, true);
+    registerIntOption_("theoretical_fragment_ions", "<num>", 0, "theoretical fragment ion peak representation, 0==sum of intensites plus flanking bins, 1==sum of intensities of central bin only", false, true);
+    registerIntOption_("use_A_ions", "<num>", 0, "use A ions for PSM, 0 == no, 1 == yes", false, true);
```

Comment:
Typically, I would suggest to use flags for binary options like this one, but the fact that the defaults are different (e.g. A ions: no, B ions: yes) doesn't fit well with that. A good alternative might be to collect all options in one string list, like we do for modifications, e.g.:
```
vector<String> default_ions = ListUtils::create<String>("B,Y");
registerStringList_("use_ions", "<ions>", default_ions, "Selection of ions to use for peptide-spectrum matching; allowed are A, B, C, X, Y, Z, and N for neutral loss ions.", false, true);
setValidStrings_("use_ions", ListUtils::create<String>("A,B,C,X,Y,Z,N"));
```